kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala

```diff
@@ -117,26 +117,45 @@ class KyuubiSessionImpl(
         openEngineSessionConf =
           optimizedConf ++ Map(KYUUBI_ENGINE_CREDENTIALS_KEY -> engineCredentials)
       }
-      val (host, port) = engine.getOrCreate(discoveryClient, extraEngineLog)
       val passwd =
         if (sessionManager.getConf.get(ENGINE_SECURITY_ENABLED)) {
           InternalSecurityAccessor.get().issueToken()
         } else {
           Option(password).filter(_.nonEmpty).getOrElse("anonymous")
         }
-      try {
-        _client = KyuubiSyncThriftClient.createClient(user, passwd, host, port, sessionConf)
-        _engineSessionHandle = _client.openSession(protocol, user, passwd, openEngineSessionConf)
-      } catch {
-        case e: Throwable =>
-          error(
-            s"Opening engine [${engine.defaultEngineName} $host:$port]" +
-              s" for $user session failed",
-            e)
-          throw e
+
+      val maxAttempts = sessionManager.getConf.get(ENGINE_OPEN_MAX_ATTEMPTS)
+      val retryWait = sessionManager.getConf.get(ENGINE_OPEN_RETRY_WAIT)
```

Comment:
It is better that we can set the max attempts and retry wait for session level?
Now, we can only set it in server side.